packages/server/lib/factory/IMAInternalFactory.js

```diff
@@ -0,0 +1,258 @@
+    res.$IMA = res.$IMA ? { ...res.$IMA, routeName } : { routeName };
+  }
+
+  function _importAppMainSync({ environment, context = {} }) {
+    let appMain = serverGlobal.has(GLOBAL.APP_MAIN)
+      ? serverGlobal.get(GLOBAL.APP_MAIN)
+      : appFactory();
+
+    if (environment.$Env === 'dev') {
+      appMain = appFactory();
+
+      instanceRecycler.clear();
+    }
+
+    if (!instanceRecycler.isInitialized()) {
+      serverGlobal.set(GLOBAL.APP_MAIN, appMain);
+      serverGlobal.set(GLOBAL.DUMMY_APP, _createDummyApp({ environment }));
+
+      instanceRecycler.init(
+        appMain.ima.createImaApp,
+        environment.$Server.concurrency
+      );
+    }
+
+    context.appMain = appMain;
+
+    return context.appMain;
+  }
+
+  function createBootConfig(event) {
+    const { req, res, environment } = event;
+    let language = res.locals.language;
+    let languagePartPath = res.locals.languagePartPath;
+    let host = res.locals.host;
+    let root = res.locals.root;
+    let urlPath = res.locals.path;
+    let protocol = res.locals.protocol;
+
+    let dictionary = languageLoader(language);
+
+    event.context.bootConfig = {
+      services: {
+        request: req,
+        response: res,
+        $IMA: {},
+        dictionary: {
+          $Language: language,
+          dictionary: dictionary,
+        },
+        router: {
+          $Protocol: protocol,
+          $Host: host,
+          $Path: urlPath,
+          $Root: root,
+          $LanguagePartPath: languagePartPath,
+        },
+      },
+      settings: {
+        $Debug: environment.$Debug,
+        $Env: environment.$Env,
+        $Version: environment.$Version,
+        $App: environment.$App || {},
+        $Source: environment.$Source,
+        $Protocol: protocol,
+        $Language: language,
+        $Host: host,
+        $Path: urlPath,
+        $Root: root,
+        $LanguagePartPath: languagePartPath,
+      },
+    };
+
+    event = emitter.emit(Event.CreateBootConfig, event);
+
+    if (event.$result) {
+      event.$result = { ...event.context.bootConfig, ...event.$result };
+      event.context.bootConfig = event.$result;
+    }
+
+    return event.context.bootConfig;
+  }
+
+  function _initApp(event) {
+    let { context } = event;
+    let bootConfig = createBootConfig(event);
+    context.app = instanceRecycler.getInstance();
+
+    emitter.emit(Event.CreateImaApp, event);
+
+    Object.assign(
+      bootConfig,
+      context.appMain.getInitialAppConfigFunctions(),
+      context.appMain.ima.getInitialPluginConfig(),
+      context.appMain.ima.getInitialImaConfigFunctions()
+    );
+
+    context.app.bootstrap.run(bootConfig);
+
+    return context.app;
+  }
+
+  function _generateAppResponse({ context }) {
+    let router = context.app.oc.get('$Router');
+
+    return router.route(router.getPath());
+  }
+
+  return {
+    _initApp,
+    createBootConfig,
+    _importAppMainSync,
+    _createDummyApp,
+    _getRouteInfo,
+    _addImaToResponse,
```

Comment:
why do these have `_` to discourage user from using it outside of the server pkg?

Reply:
only for defined interal functions